### ISSUE TITLE
Segmentation fault with clickable statusline

### ISSUE THREAD

**Issue by siduck** [neovim/neovim#21218]
### Neovim version (nvim -v)

v0.8.1

### Vim (not Nvim) behaves the same?

I dont think so, since we dont have this feature for vim right

### Operating system/version

Gnu / Void Linux

### Terminal name/version

St

### $TERM environment variable

st-256color

### Installation

system repo 

### How to reproduce the issue

minimal init.lua

```lua
vim.cmd "function! TbNewTab(a,b,c,d) \n tabnew \n endfunction"
vim.cmd "function! TbGotoTab(tabnr,b,c,d) \n execute a:tabnr ..'tabnext' \n endfunction"

local fn = vim.fn

Tabs = function()
  local result, number_of_tabs = "", fn.tabpagenr "$"
    for i = 1, number_of_tabs, 1 do
      result = result .. ("%" .. i .. "@TbGotoTab@" .. " " .. i .. " ")
      result = (i == fn.tabpagenr() and result) or result
    end

    local new_tabtn =  "%@TbNewTab@  %X"

    return  new_tabtn .. result
end

vim.opt.statusline = "%!v:lua.Tabs()"
```

### Expected behavior

to not segfault

### Actual behavior

When clicking the clickable texts in the statusline, neovim randomly segfaults and then force quits itself.

https://user-images.githubusercontent.com/59060246/204324355-4cfe25ca-d5b7-4677-b9d4-9d9d6aa6cc27.mp4

**Comment by zeertzjq** [comment]
Duplicate of #20780

**Comment by siduck** [comment]
@zeertzjq is the fix added in the recent 0.8.1 stable release or just the master?

**Comment by zeertzjq** [comment]
Just on master. It isn't very easy to backport.

**Comment by siduck** [comment]
> Just on master. It isn't very easy to backport.

oh cool! so it would be included in next release ( 0.8.2 ) right?

**Comment by zeertzjq** [comment]
v0.8.1 is not released from the master branch, neither will v0.8.2 be. It will be in v0.9.0.

**Comment by siduck** [comment]
hi @zeertzjq  0.8.2 has been released today, is this issue fixed on that release?

**Comment by clason** [comment]
No, as he explained: the fix will only be in 0.9.0.
